packages/network-controller/src/NetworkController.ts

```diff
@@ -276,7 +311,7 @@ export class NetworkController extends BaseControllerV2<
   /**
    * Ethereum provider object for the current network
    */
-  provider: any;
+  provider: ProviderEngine | undefined;
```

Comment:
Is this is a breaking change? In practice this has always been a provider or undefined. But considering that `provider` is a public property I can see how this could break people's code if they were expecting this to be typed as `any` and now it's not.

Reply:
Yeah... seems like a breaking change to me.

Reply:
Alright. I'll see if I can pull this out to a separate PR.

Reply:
I kept the type augmentations for `web3-provider-engine` as they are useful for mocking purposes in tests, but I removed the type on `provider` here: [c2c5ef9](https://github.com/MetaMask/core/pull/1039/commits/c2c5ef9c0846f20482c7e760c329dbb904130118)